addon/-private/factory-for.js

```diff
@@ -2,13 +2,22 @@
   Utility to use `ApplicationInstance#factoryFor` if available (2.12+) and
   fallback to private `_lookupFactory` otherwise
 */
+class LegacyFactoryFor {
```

Comment:
Just mimics the same public API shape, not meant to be a polyfill, but instead just provides the same expected shape now with the factoryFor methods/properties we rely on downstream within addon/validations/factory.js